src/main/java/org/wildfly/extension/elytron/SaslServerDefinitions.java

```diff
@@ -221,6 +222,10 @@ static ResourceDefinition getConfigurableSaslServerFactoryDefinition() {
                     Predicate<String> filter = null;
                     List<ModelNode> nodes = model.require(ElytronDescriptionConstants.FILTERS).asList();
                     for (ModelNode current : nodes) {
+                        if (current.hasDefined(ElytronDescriptionConstants.PREDEFINED_FILTER)
+                                && current.hasDefined(ElytronDescriptionConstants.PATTERN_FILTER)) {
+                            throw ROOT_LOGGER.invalidDefinition(ElytronDescriptionConstants.FILTERS);
+                        }
```

Comment:
Is there any way we can get this validation to happen earlier?

This method is only executed during Stage.RUNTIME when really we should be able to validate this during the model stage?

Don't spend too much time on it if it is very complex as we need to simplify this resource anyway.